python/lsst/sims/photUtils/SedList.py

```diff
@@ -0,0 +1,391 @@
+                fNorm = sed.calcFluxNorm(magNorm, self._imsimband)
+                sed.multiplyFluxNorm(fNorm)
+
+            temp_sed_list.append(sed)
+
+
+        if internalAvList is not None:
+            self._av_int_wavelen, \
+            self._a_int, \
+            self._b_int = self.applyAv(temp_sed_list, internalAvList,
+                                       self._av_int_wavelen, self._a_int, self._b_int)
+
+        if redshiftList is not None:
+            self.applyRedshift(temp_sed_list, redshiftList)
+
+        if self._wavelen_match is not None:
+            for sedObj in temp_sed_list:
+                if sedObj.wavelen is not None:
+                    sedObj.resampleSED(wavelen_match=self._wavelen_match)
+
+        if galacticAvList is not None:
+            self._av_gal_wavelen, \
+            self._a_gal, \
+            self._b_gal = self.applyAv(temp_sed_list, galacticAvList,
+                                       self._av_gal_wavelen, self._a_gal, self._b_gal)
+
+        self._sed_list += temp_sed_list
+
+        self._initialized = True
+
+
+
+    def applyAv(self, sedList, avList, dustWavelen, aCoeffs, bCoeffs):
+        """
+        Take the array of Sed objects sedList and apply extinction due to dust.
+
+        This method makes the necessary changes to the Seds in SedList in situ.
+        It returns the wavelength grid and corresponding dust coefficients so that
+        they an be reused on Seds with identical wavelength grids.
+
+        @param [in] sedList is a list of Sed objects
+
+        @param [in] avList is a list of Av extinction values internal to each object
+
+        @param [in] dustWavelen is the wavelength grid corresponding to the
+        dust model coefficients.  If this differs from the wavelength grid
+        of any of the Seds in sedList, the dust model coefficients will be
+        re-generated.
+
+        @param [in] aCoeffs are the 'a' dust model coefficients (see O'Donnell 1994
+        ApJ 422 158)
+
+        @param [in] bCoeffs are the 'b' dust model coefficients from O'Donnell.
+
+        @param [out] dustWavelen as generated/used by this method
+
+        @param [out] aCoeffs as generated/used by this method
+
+        @param [out] bCoeffs as generated/used by this method
+
+        aCoeffs and bCoeffs are re-generated as needed
+        """
+
+        for sedobj, av in zip(sedList, avList):
+            if sedobj.wavelen is not None:
+                #setupCCMab only depends on the wavelen array
+                #because this is supposed to be the same for every
+                #SED object in sedList, it is only called once for
+                #each invocation of applyAv
+
+                if dustWavelen is None or len(sedobj.wavelen)!=len(dustWavelen) \
+                or (sedobj.wavelen!=dustWavelen).any():
+                    aCoeffs, bCoeffs = sedobj.setupCCMab()
+                    dustWavelen = sedobj.wavelen
+
+                sedobj.addCCMDust(aCoeffs, bCoeffs, A_v=av)
```

Comment:
Yes, but that should be a separate issue.  I assume that there is a reason the name did not get changed when we moved to O'Donnell dust initially.  I want to talk to Lynne to see what that reason was.